spec/requests/posts_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+
+RSpec.describe 'posts', type: :request do
+  describe 'GET /posts' do
+    it 'works! returns correct status code' do
+      get user_posts_path(user_id: 1)
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'works!  renders index view correctly' do
+      get user_posts_path(user_id: 1)
+      expect(response).to render_template(:index)
+    end
+
+    it 'works! reponse body includes <h1>Posts#index<h1>' do
+      get user_posts_path(user_id: 1)
+      expect(response.body).to include('<h1>Posts#index<h1>')
+    end
+  end
+
+  describe 'GET /posts/:id' do
+    it 'works! returns correct status code' do
+      get user_post_path(1, id: 1)
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'works!  renders show view correctly' do
+      get user_post_path(1, id: 1)
+      expect(response).to render_template(:show)
+    end
+
+    it 'works! reponse body includes <h1>Posts#index</h1> ' do
+      get user_post_path(1, id: 1)
+      expect(response.body).to include('<h1>Posts#index</h1>')
+    end
+  end
+end
```

Comment:
- [ ] Some of your tests are failing please fix them
![fail](https://user-images.githubusercontent.com/44745855/149992989-f48729f0-49a2-45c9-8d1e-4ab96c31810f.png)